pkg/cmd/server/origin/run_components.go

```diff
@@ -367,13 +363,116 @@ func (c *MasterConfig) RunDeploymentConfigController() {
 func (c *MasterConfig) RunDeploymentTriggerController() {
 	dcInfomer := c.Informers.DeploymentConfigs().Informer()
 	rcInformer := c.Informers.ReplicationControllers().Informer()
-	streamInformer := c.Informers.ImageStreams().Informer()
 	osclient := c.DeploymentTriggerControllerClient()
 
-	controller := triggercontroller.NewDeploymentTriggerController(dcInfomer, rcInformer, streamInformer, osclient, c.ExternalVersionCodec)
+	controller := triggercontroller.NewDeploymentTriggerController(dcInfomer, rcInformer, nil, osclient, c.ExternalVersionCodec)
 	go controller.Run(5, utilwait.NeverStop)
 }
 
+// TODO: remove when generated informers exist
+type temporaryLister struct {
+	*oscache.StoreToImageStreamLister
+}
+
+func (l temporaryLister) ImageStreams(namespace string) imagetriggercontroller.ImageStreamNamespaceLister {
+	return l.StoreToImageStreamLister.ImageStreams(namespace)
+}
+
+type podSpecUpdater struct {
+	kclient kclientset.Interface
+}
+
+func (u podSpecUpdater) Update(obj runtime.Object) error {
+	switch t := obj.(type) {
+	case *kextensions.DaemonSet:
+		_, err := u.kclient.Extensions().DaemonSets(t.Namespace).Update(t)
+		return err
+	case *kextensions.Deployment:
+		_, err := u.kclient.Extensions().Deployments(t.Namespace).Update(t)
+		return err
+	case *kapps.StatefulSet:
+		_, err := u.kclient.Apps().StatefulSets(t.Namespace).Update(t)
+		return err
+	case *kapi.Pod:
+		_, err := u.kclient.Core().Pods(t.Namespace).Update(t)
+		return err
+	default:
+		return fmt.Errorf("unrecognized object - no trigger update possible for %T", obj)
+	}
+}
+
+func (c *MasterConfig) RunImageTriggerController() {
+	streamInformer := c.Informers.ImageStreams().Informer()
+	lister := temporaryLister{c.Informers.ImageStreams().Lister()}
+
+	oclient, kclient := c.ImageTriggerControllerClients()
+	updater := podSpecUpdater{kclient}
+	bcInstantiator := buildclient.NewOSClientBuildConfigInstantiatorClient(oclient)
+	broadcaster := imagetriggercontroller.NewTriggerEventBroadcaster(kclient.Core())
+
+	sources := []imagetriggercontroller.TriggerSource{
+		{
+			Resource:  unversioned.GroupResource{Group: "build.openshift.io", Resource: "buildconfigs"},
```

Comment:
This isn't registration, this is a unique prefix and message path.  It's there to ensure trigger sources are unique.